File: backend/src/main/java/com/service/backend/auth/presentation/dto/RegisterRequest.java
package com.service.backend.auth.presentation.dto;

import jakarta.validation.constraints.*;
import lombok.Data;

import java.time.LocalDate;

@Data
public class RegisterRequest {
    @NotBlank(message = "Email is required")
    @Email(message = "Invalid email format")
    private String email;

    @NotBlank(message = "Password is required")
    @Size(min = 8, message = "Password must be at least 8 characters long")
    @Pattern(
            regexp="^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).+$",
            message="Password must contain at least one uppercase letter, one lowercase letter, and one number"
    )
    private String password;

//    @NotBlank(message = "fullName is required")
//    private String fullName;

//     @NotBlank(message = "reCAPTCHA token is required")
//     private String recaptchaToken;

//     @Pattern(
//             regexp="^(MALE|FEMALE|OTHER)$",
//             message="Gender must be one of MALE, FEMALE, OTHER"
//     )
//     private String gender;

//     @Pattern(
//             regexp="^(\\+84|0)[0-9]{9,10}$",
//             message="Phone number is not valid"
//     )
//     private String phone;

//     @NotNull(message = "Date of birth is required")
//     @Past(message = "Date of birth must be in the past")
//     private LocalDate dob;


//     private String bio;
}

File: mobile_flutter/lib/features/auth/data/models/register_request.dart
/// Registration payload. Maps to the backend `RegisterRequest`:
/// `studentId` carries the student id. `organizationId` is required.
class RegisterRequest {
  final String email;
  final String studentId;
  final String fullName;
  final String password;
  final int organizationId;

  const RegisterRequest({
    required this.email,
    required this.studentId,
    required this.fullName,
    required this.password,
    required this.organizationId,
  });

  Map<String, dynamic> toJson() => {
        'email': email,
        'studentId': studentId,
        'fullName': fullName,
        'password': password,
        'organizationId': organizationId,
      };
}
